[Auto-generated] java.lang.NoClassDefFoundError: com/intellij/mock/MockProjectEx

# Version

6.5.0


# Exception


## Message

com/intellij/mock/MockProjectEx


## Stacktrace

```
java.lang.NoClassDefFoundError: com/intellij/mock/MockProjectEx
	at org.elixir_lang.parser.ExternalRules.ifVersion(ExternalRules.java:18)
	at org.elixir_lang.parser.ElixirParser.unmatchedUnaryNonNumericOperation_0(ElixirParser.java:6528)
	at org.elixir_lang.parser.ElixirParser.unmatchedUnaryNonNumericOperation(ElixirParser.java:6516)
	at org.elixir_lang.parser.ElixirParser.unmatchedExpression(ElixirParser.java:6325)
	at org.elixir_lang.parser.ElixirParser.expression(ElixirParser.java:2028)
	at org.elixir_lang.parser.ElixirParser.expressionList(ElixirParser.java:2040)
	at org.elixir_lang.parser.ElixirParser.elixirFile_1_0(ElixirParser.java:1913)
	at org.elixir_lang.parser.ElixirParser.elixirFile_1(ElixirParser.java:1904)
	at org.elixir_lang.parser.ElixirParser.elixirFile(ElixirParser.java:1889)
	at org.elixir_lang.parser.ElixirParser.parse_root_(ElixirParser.java:479)
	at org.elixir_lang.parser.ElixirParser.parseLight(ElixirParser.java:473)
	at org.elixir_lang.parser.ElixirParser.parse(ElixirParser.java:20)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:88)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:49)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.ensureParsed(LazyParseableElement.java:192)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:240)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getFirstChildNode(LazyParseableElement.java:39)
	at com.intellij.psi.impl.source.tree.injected.MultiHostRegistrarImpl.keepTreeFromChameleoningBack(MultiHostRegistrarImpl.java:358)
	at com.intellij.psi.impl.source.tree.injected.MultiHostRegistrarImpl.doneInjecting(MultiHostRegistrarImpl.java:235)
	at org.intellij.plugins.markdown.injection.CodeFenceInjector.getLanguagesToInject(CodeFenceInjector.java:42)
	at com.intellij.psi.impl.source.tree.injected.InjectedPsiCachedValueProvider$MyInjProcessor.process(InjectedPsiCachedValueProvider.java:82)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageManagerImpl.processInPlaceInjectorsFor(InjectedLanguageManagerImpl.java:461)
	at com.intellij.psi.impl.source.tree.injected.InjectedPsiCachedValueProvider.doCompute(InjectedPsiCachedValueProvider.java:62)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.probeElementsUp(InjectedLanguageUtil.java:312)
	at com.intellij.psi.impl.source.tree.injected.InjectedLanguageUtil.enumerate(InjectedLanguageUtil.java:144)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.queryLineMarkersForInjected(LineMarkersPass.java:241)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.queryProviders(LineMarkersPass.java:203)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.lambda$doCollectInformation$3(LineMarkersPass.java:111)
	at com.intellij.codeInsight.daemon.impl.Divider.divideInsideAndOutsideInOneRoot(Divider.java:96)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.doCollectInformation(LineMarkersPass.java:106)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$null$1(PassExecutorService.java:438)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1160)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:431)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:548)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:493)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:94)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:430)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:406)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:142)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:249)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:404)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:165)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassNotFoundException: com.intellij.mock.MockProjectEx PluginClassLoader[org.elixir_lang, 6.5.0] com.intellij.ide.plugins.cl.PluginClassLoader@5a4221ac
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:65)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 47 more

```

--- a/src/org/elixir_lang/parser/ExternalRules.java
+++ b/src/org/elixir_lang/parser/ExternalRules.java
@@ -1,7 +1,6 @@
 package org.elixir_lang.parser;
 
 import com.intellij.lang.PsiBuilder;
-import com.intellij.mock.MockProjectEx;
 import com.intellij.openapi.project.Project;
 import org.elixir_lang.sdk.elixir.Release;
 import org.elixir_lang.sdk.elixir.Type;
@@ -15,7 +14,11 @@ public class ExternalRules {
         Project project = psiBuilder.getProject();
         Release release = Type.getRelease(project);
 
-        assert !(release == null && project instanceof MockProjectEx) :
+
+        assert !(release == null &&
+                /* `project instanceof MockProjectEx`, but in a way that safe for IDEs that don't ship with
+                   `MockProjectEx` */
+                project.getClass().getCanonicalName().equals("com.intellij.mock.MockProjectEx")) :
                 "Release MUST be set during testing of ifVersion rules:\n" +
                         "Call `setProjectSdkFromEbinDirectory();` to setup the Release.";
 

